### PR TITLE
Purge user PII data

### PR DESCRIPTION
**Pull request by hacodeorg**
[FND-1417](https://codedotorg.atlassian.net/browse/FND-1417)

Data to purge
- Contact rollups data in `ContactRollupsFinal`.
- Survey data in `Pd::MiscSurvey`, `Foorm::Submission`, and `Foorm::SimpleSurveySubmission`.
- Professional development email history in `Pd::Application::Email`.

No action needed
- The current process already purges `TeacherFeedback` and `ContactRollupsPardotMemory`.
- `ContactRollupsRaw` and `ContactRollupsProcessed` are just intermediate tables. They are dropped and rebuilt everyday by the contact rollups process.. 
- `contact_rollups_comparison` table was dropped long ago.
- `Pd::Application::Tag` model doesn't contain PII data. The `name` field is just tag name. In addition, the `pd_application_tags` table is currently empty.
- `Donor` model doesn't contain PII data. It has organization/donor names, URLs and Twitter links; all can be considered public information. Moreover, the `donors` table has only 166 records. They can be manually deleted upon request.

## Links
- [PLC PII data](https://docs.google.com/spreadsheets/d/1s_AoB4UnwKfyOLyJD2Oozg_m6NizIL15JJb9BTbPo0M/edit#gid=1738712892)
- [LP PII data](https://docs.google.com/spreadsheets/d/12UY_ye1UAxnYRG4e2mytdgBx_AxK3BFS4U7jsD8G0pY/edit#gid=0)

## Testing story
- Run unit tests `bundle exec rails test test/helpers/delete_accounts_helper_test.rb`

## Follow-up work
- [x] Drop `survey_answers` table. Last created row was from 2019-10-22. We have migrated previous survey data to the new `foorm` system.
- [x] Back-purging data in `ContactRollupsFinal`, `ContactRollupsPardotMemory`, `Pd::MiscSurvey`, `Foorm::Submission`, `Foorm::SimpleSurveySubmission`, and `Pd::Application::Email`.

## PR Checklist:
- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by bencodeorg — APPROVED** [review]

File: lib/cdo/delete_accounts_helper.rb [hunk 1]
```diff
@@ -281,10 +290,25 @@ def assert_constraint(condition, message)
...
+    @log.puts "Deleting ContactRollups records for email #{email}"
+    return unless email
+
+    # Contact rollups data are in 4 tables: ContactRollupsRaw, ContactRollupsProcessed,
+    # ContactRollupsPardotMemory, and ContactRollupsFinal.
+    # During daily contact rollups runs, ContactRollupsRaw and ContactRollupsProcessed
+    # are dropped, records marked for deletion in ContactRollupsPardotMemory are
+    # also purged. In addition, records in Pardot server are also deleted.
+    # Thus, only need to delete ContactRollupsFinal record here.
+    ContactRollupsFinal.find_by_email(email)&.destroy
```
Comment (daynew) [inline lib/cdo/delete_accounts_helper.rb:308]: Do you need to call `really_destroy`? Or set the email field to `nil`?

File: dashboard/test/helpers/delete_accounts_helper_test.rb [hunk 2]
```diff
@@ -2052,6 +2052,26 @@ def assert_bucket_hard_deletes_contents(bucket)
...
   end
 
+  test 'deletes contact rollups final data' do
+    teacher = create :teacher
+    teacher_email = teacher.email
+    create :contact_rollups_final, email: teacher_email
+
+    assert ContactRollupsFinal.find_by_email(teacher_email)
+    purge_user teacher
+    assert_nil ContactRollupsFinal.find_by_email(teacher_email)
```
Comment (daynew) [inline dashboard/test/helpers/delete_accounts_helper_test.rb:2062]: Do you also need the `.with_deleted` so it checks for deleted data?

File: dashboard/test/helpers/delete_accounts_helper_test.rb [hunk 3]
```diff
@@ -2052,6 +2052,26 @@ def assert_bucket_hard_deletes_contents(bucket)
...
+  end
+
+  test 'marks contact rollups pardot memory for deletion' do
+    teacher = create :teacher
+    teacher_email = teacher.email
+    create :contact_rollups_pardot_memory, email: teacher_email
+
+    assert_nil ContactRollupsPardotMemory.find_by_email(teacher_email).marked_for_deletion_at
+    purge_user teacher
+    refute_nil ContactRollupsPardotMemory.find_by_email(teacher_email).marked_for_deletion_at
```
Comment (daynew) [inline dashboard/test/helpers/delete_accounts_helper_test.rb:2072]: Same comment to check with `.with_deleted`

File: lib/cdo/delete_accounts_helper.rb (hunk 1 above)
Comment (hacodeorg) [reply lib/cdo/delete_accounts_helper.rb:308]: ContactRollupsFinal does not use paranoia features. `destroy` will delete records from database (after triggering model hooks).

File: dashboard/test/helpers/delete_accounts_helper_test.rb (hunk 2 above)
Comment (hacodeorg) [reply dashboard/test/helpers/delete_accounts_helper_test.rb:2062]: ContactRollupsFinal and ContactRollupsPardotMemory don't use paranoia so they don't have `.with_deleted`.

**Review by daynew — APPROVED** [review]

**Review by sureshc — APPROVED** [review]

**Pull request merged**